### PR TITLE
JS: add a js/empty-password-in-configuration-file query

### PR DESCRIPTION
**Pull request by erik-krogh**
[Evaluation looks clean](https://github.com/github/codeql-dca-main/tree/data/erik-krogh/CWE-862__default__CustomSuite/reports). 

[Inspired by this C# query](https://github.com/github/codeql/blob/main/csharp/ql/src/Configuration/EmptyPasswordInConfigurationFile.ql).  
The implementation mostly reuses the logic from our existing `PasswordInConfigurationFile.ql` query.

### CODE REVIEW

File: javascript/ql/src/Security/CWE-862/EmptyPasswordInConfigurationFile.qhelp [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<!DOCTYPE qhelp PUBLIC
+  "-//Semmle//qhelp//EN"
+  "qhelp.dtd">
+<qhelp>
+<overview>
+<p>The use of an empty string as a password in a configuration file is not secure.</p>
+
+</overview>
+<recommendation>
+<p>Choose a proper password and encrypt it if you need to store it in the configuration file.</p>
```
Comment (esbena) [inline javascript/ql/src/Security/CWE-862/EmptyPasswordInConfigurationFile.qhelp:10]:
```suggestion
<p>Choose a proper password and encrypt it if it has to stored in a configuration file.</p>
```

**Review by esbena — COMMENTED** [review]
I think the qhelp has the right length. It would be too patronising to add examples for this kind of security issue.

**Review by esbena — DISMISSED** [review]

**Review by felicitymay — COMMENTED** [review]
Text generally looks okay but there seem to be some inconsistencies, which I've commented on.

File: javascript/ql/src/Security/CWE-862/EmptyPasswordInConfigurationFile.qhelp [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<!DOCTYPE qhelp PUBLIC
+  "-//Semmle//qhelp//EN"
+  "qhelp.dtd">
+<qhelp>
+<overview>
+<p>The use of an empty string as a password in a configuration file is not secure.</p>
+
+</overview>
+<recommendation>
+<p>Choose a proper password and encrypt it if it has to stored in a configuration file.</p>
```
Comment (kaeluka) [inline javascript/ql/src/Security/CWE-862/EmptyPasswordInConfigurationFile.qhelp:10]:
```suggestion
<p>Choose a proper password and encrypt it if it has to be stored in a configuration file.</p>
```

**Review by esbena — APPROVED** [review]

**Review by felicitymay — APPROVED** [review]
Thanks for the updates

**Pull request merged**